src/components/posts/PostList.js

```diff
@@ -1,142 +1,146 @@
-import { useEffect, useState } from "react"
-import { getPosts, getPostsByCategory, getPostsByUser } from "../../managers/posts"
-import { getUsers } from "../../managers/users"
-import { getCategories } from "../../managers/categories"
-import { Link} from "react-router-dom"
+import { useEffect, useState } from "react";
+import { getPosts, getPostsByCategory, getPostsByTag, getPostsByTitle, getPostsByUser } from "../../managers/posts";
```

Comment:
If you are doing all filtering on client side, you don't need `getPostsByCategory, getPostsByTag, getPostsByTitle, getPostsByUser`. If you do want to use these functions, you can adjust the logic and find the common posts among all of the filtering options. Or we don't even have to worry about stacking the filtering options if we want to keep it simple!